components/HTMLBlock/index.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useEffect } from 'react';
+import { renderToStaticMarkup } from 'react-dom/server';
+
+const MATCH_SCRIPT_TAGS = /<script\b[^>]*>([\s\S]*?)<\/script *>\n?/gim;
+
+const extractScripts = (html: string = ''): [string, () => void] => {
+  const scripts: string[] = [];
+  let match: RegExpExecArray | null;
+  while ((match = MATCH_SCRIPT_TAGS.exec(html)) !== null) {
+    scripts.push(match[1]);
+  }
+  const cleaned = html.replace(MATCH_SCRIPT_TAGS, '');
+  return [cleaned, () => scripts.map(js => window.eval(js))];
+};
+
+const HTMLBlock = ({ children = '', runScripts = false, safeMode = false }) => {
+  let html = children;
+  if (typeof html !== 'string') html = renderToStaticMarkup(html);
+  const [cleanedHtml, exec] = extractScripts(html);
+
+  useEffect(() => {
+    if (typeof window !== 'undefined' && typeof runScripts === 'boolean' && runScripts) exec();
+  }, [runScripts, exec]);
+
+  if (safeMode) {
```

Comment:
`safeMode` was my silly attempt to make suggested edits safer. Suggested edits could be submitted by anonymous users. And it features a preview mode, so if the submitter wasn't an admin, we wouldn't render the HTML. 🤷🏼

Reply:
that was the usage for us sure, but i'm wondering about the 1000 or downloads a week of `@readme/markdown`? is this new improved renderer going to be internal only or are we improving on the current one?

Reply:
okay, we can figure out the final details in another PR since it's a bit out of scope for this one 😅